app/helpers/agent_helper.rb

```diff
@@ -82,6 +82,12 @@ def agent_type_icon(agent, agents)
     end
   end
 
+  def agent_type_select_options
+    Rails.cache.fetch('agent_type_select_options') do
+      Agent.types.map {|type| [agent_type_to_human(type.name), type, {title: h(Agent.build_for_type(type.name, User.new(id: 0), {}).html_description.lines.first.strip)}] }
```

Comment:
The descriptions could have sensitive information in them, but they do not at the moment. I am not passing the `current_user` avoid the risk of leaking and checked the first line of the current agent descriptions to make sure they do not include user specific content.
To completely avoid the problem we would need to cache the data per user (which would increase the memory consumption on multi-user environments).